source/dem/insertion.cc

```diff
@@ -76,19 +76,21 @@ Insertion<dim>::assign_particle_properties(
         -particle_sizes[particle_counter];
       double density =
         physical_properties.density_particle[current_inserting_particle_type];
-      double vel_x                   = dem_parameters.insertion_info.vel_x;
-      double vel_y                   = dem_parameters.insertion_info.vel_y;
-      double vel_z                   = dem_parameters.insertion_info.vel_z;
-      double omega_x                 = dem_parameters.insertion_info.omega_x;
-      double omega_y                 = dem_parameters.insertion_info.omega_y;
-      double omega_z                 = dem_parameters.insertion_info.omega_z;
-      double fem_force_x             = 0.;
-      double fem_force_y             = 0.;
-      double fem_force_z             = 0.;
-      double volumetric_contribution = 0.;
-      double mass                    = density * 4. / 3. * M_PI *
+      double vel_x        = dem_parameters.insertion_info.vel_x;
+      double vel_y        = dem_parameters.insertion_info.vel_y;
+      double vel_z        = dem_parameters.insertion_info.vel_z;
+      double omega_x      = dem_parameters.insertion_info.omega_x;
+      double omega_y      = dem_parameters.insertion_info.omega_y;
+      double omega_z      = dem_parameters.insertion_info.omega_z;
+      double fem_force_x  = 0.;
+      double fem_force_y  = 0.;
+      double fem_force_z  = 0.;
+      double fem_torque_x = 0.;
+      double fem_torque_y = 0.;
+      double fem_torque_z = 0.;
```

Comment:
So question, why do we have fem_force and fem_torque as a bunch of doubles. Why not make it a Tensor<1,dim> and be done with it?

Reply:
Because of the vector of double at line 95 ?

Reply:
Yes, because it will be stored in a vector.